test/Mono.Linker.Tests.Cases/Inheritance.Interfaces/StaticInterfaceMethods/StaticAbstractInterfaceMethodsLibrary.cs

```diff
@@ -0,0 +1,897 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Mono.Linker.Tests.Cases.Expectations.Assertions;
+using Mono.Linker.Tests.Cases.Expectations.Helpers;
+using Mono.Linker.Tests.Cases.Expectations.Metadata;
+
+namespace Mono.Linker.Tests.Cases.Inheritance.Interfaces.StaticInterfaceMethods
+{
+	[SetupLinkerArgument ("-a", "test.exe", "library")]
+	public static class StaticAbstractInterfaceMethodsLibrary
+	{
+		public static void Main ()
+		{
+			InterfaceMethodsUsedThroughConstrainedType.Test ();
+			InterfaceWithMethodsUsedEachWay.Test ();
+			InterfaceMethodUsedOnConcreteType.Test ();
+			InterfaceMethodsKeptThroughReflection.Test ();
+			StaticInterfaceInheritance.Test ();
+			GenericStaticInterface.Test ();
+			RecursiveGenericInterface.Test ();
+			UnusedInterfaces.Test ();
+		}
+
+		[Kept]
+		public static class InterfaceMethodsUsedThroughConstrainedType
+		{
+			[Kept]
+			public interface IUsedThroughConstrainedType
+			{
+				[Kept]
+				static abstract int UsedThroughConstrainedType ();
+			}
+
+			[Kept]
+			internal interface IUsedThroughConstrainedTypeInternal
+			{
+				static abstract int UsedThroughConstrainedType ();
+			}
+
+			[Kept]
+			[KeptMember (".ctor()")]
+			[KeptInterface (typeof (IUsedThroughConstrainedType))]
+			[KeptInterface (typeof (IUsedThroughConstrainedTypeInternal))]
+			public class UsesIUsedThroughConstrainedTypeMethods : IUsedThroughConstrainedType, IUsedThroughConstrainedTypeInternal
+			{
+				[Kept]
+				[KeptOverride (typeof (IUsedThroughConstrainedType))]
+				[RemovedOverride (typeof (IUsedThroughConstrainedTypeInternal))]
+				public static int UsedThroughConstrainedType () => 0;
+			}
+
+			[Kept]
+			[KeptMember (".ctor()")]
+			[KeptInterface (typeof (IUsedThroughConstrainedType))]
+			[KeptInterface (typeof (IUsedThroughConstrainedTypeInternal))]
+			public class UnusedIUsedThroughConstrainedTypeMethods : IUsedThroughConstrainedType, IUsedThroughConstrainedTypeInternal
+			{
+				[Kept]
+				[KeptOverride (typeof (IUsedThroughConstrainedType))]
+				[RemovedOverride (typeof (IUsedThroughConstrainedTypeInternal))]
+				public static int UsedThroughConstrainedType () => 0;
+			}
+
+			// Should this be kept?
+			private class UnusedIUsedThroughConstrainedTypeMethodsPrivate : IUsedThroughConstrainedType, IUsedThroughConstrainedTypeInternal
```

Comment:
I couldn't find any references to it, so it seems like it should be removed. Why would it be kept?

Reply:
If I remember my thought process, I was wondering if the static interface method being kept would cause this to be kept. I don't think that makes sense and I agree it should be removed though.